add initial content to Documentation index (#20)
elmiko/ocp-infrastructure-provider-onboarding-guide: docs/documentation/index.md

# Documentation

All OpenShift components that are released from Red Hat will have accompanying
documentation in the [OpenShift product documentation][ocp-docs]. The content
for these product documents are created and maintained by Red Hat, but they
are inspired by the repository level documentation associated with the component.

## Repository level documentation

> It is expected that any component which has its code in a Git repository will have
> documentation in that repository to cover its usage and troubleshooting. This

Reviewer: If you do want to make this more of a hard requirement, perhaps:
```suggestion
Any component that has its code in a Git repository must have
documentation in that repository to cover its usage and troubleshooting. This
```

Author: i like that, thanks

# Documentation

All OpenShift components that are released from Red Hat will have accompanying
documentation in the [OpenShift product documentation][ocp-docs]. The content
for these product documents are created and maintained by Red Hat, but they
are inspired by the [OpenShift enhancement](../openshift-enhancement),

> repository level documentation, and direct discussions with the component

Reviewer 2: Hm, I'm thinking about some examples of good repo level documentation which we like, would nice be refer something as example.

Author: i think that's a good idea, was there an example doc that you were thinking about?

we could point to the MAO documentation, but that's quite verbose. perhaps the [alibaba readme](https://github.com/openshift/cluster-api-provider-alibaba/blob/main/README.md), that seems to have some nice info about how to deploy and use the controller. wdyt?

Author: maybe [cloud provider ibm](https://github.com/openshift/cloud-provider-ibm/blob/master/README.md) too?

Reviewer 2: Alibaba looks nice, lets reference it there?

Our MAPI docs is also good enough example i think

Author: sounds good, i'll add some link. thanks!
